### ISSUE TITLE
explain better what the pull method should do in the underlying source API

### ISSUE THREAD

**Issue by skybrian** [whatwg/streams#1228]
The specification of the pull method in [4.2.3](https://streams.spec.whatwg.org/#underlying-source-api) doesn't actually say how it's supposed to return new chunks. From [example 10.4](https://streams.spec.whatwg.org/#example-rs-pull) it seems that the way to return new chunks is to call controller.enqueue(), which is the same as a push source. But it should probably say so explicitly?

Or, possibly this could be explained in the [mdn documentation](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/ReadableStream) if it doesn't belong in the spec. As it is, how to implement a push method is somewhat mysterious.

**Comment by domenic** [comment]
Do any of the examples in https://streams.spec.whatwg.org/#creating-examples help?

**Comment by skybrian** [comment]
Yes, the examples are good. Looking at it again, I think it's more an issue with the mdn documentation, which has no pull examples.

**Comment by skybrian** [comment]
On second thought, I'm not sure how to contribute a patch, but here's what I have in mind.

> A call to the pull method signals that more chunks should be enqueued using the controller's enqueue() method. Typically this happens asynchronously, and may happen indirectly in the case of push sources.

A slight ambiguity here: I believe that start() and pull() always receive the same controller and it doesn't matter which one you call enqueue() on. Is this actually specified, though?

**Comment by domenic** [comment]
Yep, that's specified! E.g. https://streams.spec.whatwg.org/#set-up-readable-stream-default-controller-from-underlying-source passes (well, closes over) the same _controller_ variable for both `start` and `pull`.
